### PR TITLE
Fix chalice deployment bug introduced in #270

### LINKED ISSUE #315: Bug introduced in #270 causes Chalice deployments to produce failing lambda function
#270 introduced a bug resulting in failure to deploy and run when using Chalice. The cause is that the Chalice packager does not include the full contents of the chalice wheel/package in the generated deployment zip. Code that was added to support local development imports `from chalice.config import Config` which fails in a production deployment with the error `[ERROR] Runtime.ImportModuleError: Unable to import module 'app': No module named 'chalice.config'`. The solution is to refactor the LocalLambdaClient into a separate file and wrap the import in try/catch when in production.
 
### Reproducible in:

slack_bolt==1.5.0
Python 3.8.6
AWS Lambda Python 3.8

#### Steps to reproduce:

1. `chalice package` or `chalice deploy`

### Expected result:

Deployments through `chalice` should produce working lambda function.

### Actual result:

Deployed lambda function fails with error:
`[ERROR] Runtime.ImportModuleError: Unable to import module 'app': No module named 'chalice.config'`

### PR DESCRIPTION
**Pull request by jlujan-invitae**
Fix chalice deployment bug introduced in #270, closes #315

* Refactor LocalLambdaClient into seperate file
* try/catch import when running from a deployed lambda

### Category

* [ ] `slack_bolt.App` and/or its core components
* [ ] `slack_bolt.async_app.AsyncApp` and/or its core components
* [x] Adapters in `slack_bolt.adapter`
* [ ] Document pages under `/docs`
* [ ] Others

## Requirements (place an `x` in each `[ ]`)

Please read the [Contributing guidelines](https://github.com/slackapi/bolt-python/blob/main/.github/contributing.md) and [Code of Conduct](https://slackhq.github.io/code-of-conduct) before creating this issue or pull request. By submitting, you are agreeing to those rules.

* [x] I've read and understood the [Contributing Guidelines](https://github.com/slackapi/bolt-python/blob/main/.github/contributing.md) and have done my best effort to follow them.
* [x] I've read and agree to the [Code of Conduct](https://slackhq.github.io/code-of-conduct).
* [x] I've run `./scripts/install_all_and_run_tests.sh` after making the changes.

### CODE REVIEW

**Review by seratch — CHANGES_REQUESTED** [review]
Thanks for submitting the changes! Can you check my comments?

File: slack_bolt/adapter/aws_lambda/chalice_handler.py [hunk 1]
```diff
@@ -16,39 +14,23 @@
...
-        return InvokeResponse(payload=response)
+try:
+    from slack_bolt.adapter.aws_lambda.local_lambda_client import LocalLambdaClient
+except ImportError:
+    LocalLambdaClient = None
 
 
 class ChaliceSlackRequestHandler:
-    def __init__(self, app: App, chalice: Chalice):  # type: ignore
+    def __init__(self, app: App, chalice: Chalice, lambda_client: Optional[Any] = None):  # type: ignore
```
[the remark this answers is not in the document]
Comment (jlujan-invitae) [reply slack_bolt/adapter/aws_lambda/chalice_handler.py:24]: The actual type is runtime generated by botocore from an API spec. It has a base of botocore.client.BaseClient. I will use that. However, it provides no real type information. Will also  add a docstring with more details.

[the remark this answers is not in the document]
Comment (seratch) [reply slack_bolt/adapter/aws_lambda/chalice_handler.py:24]: Looks good to me. I was also thinking we can use `BaseClient` at least.

**Comment by jlujan-invitae** [comment]
I identified how this bug went undetected and wrote a description in an issue with the aws/chalice project. The gist is that I was using a specific git revision from my fork in our requirements.txt. Apparently, the chalice packer ignores source distribution and attempts to download a binary distribution by package name and version. Since my fork was pre-release, it detected the previous slack-bolt version and downloaded the available wheel from PyPI. Relevant issue is here https://github.com/aws/chalice/issues/1700#issuecomment-829703418

File: tests/adapter_tests/aws/test_aws_chalice.py [hunk 2]
```diff
@@ -315,13 +315,72 @@ def events() -> Response:
...
 
+    @mock.patch(
+        "slack_bolt.adapter.aws_lambda.chalice_lazy_listener_runner.boto3",
+        autospec=True,
+    )
+    def test_lazy_listeners_non_cli(self, mock_boto3):
+        with mock.patch.dict(os.environ, {"AWS_CHALICE_CLI_MODE": "false"}):
+            assert os.environ.get("AWS_CHALICE_CLI_MODE") == "false"
+
+            mock_s3 = mock.MagicMock()
```
Comment (seratch) [inline tests/adapter_tests/aws/test_aws_chalice.py:333]: I wanted to make sure if this is for s3 client. Is it a lambda client? (I might be missing something)

**Review by seratch — COMMENTED** [review]
Thanks for updating this. Apart from my comment about the test code, everything looks great to me. Can you check my comment?

File: tests/adapter_tests/aws/test_aws_chalice.py (hunk 2 above)
Comment (jlujan-invitae) [reply tests/adapter_tests/aws/test_aws_chalice.py:333]: My bad. It is the lambda client. Will fix.

**Review by seratch — APPROVED** [review]
LGTM - thanks!

**Pull request merged**